travis.sh

```diff
@@ -21,35 +20,32 @@ upstream()
     packages/upstream               \
     packages/upstream-extra         \
     -maxdepth 1 -mindepth 1 -type d \
-  | awk -F/ '{print $NF}'           \
-  | egrep -v '^(systemd)'
+    | awk -F/ '{print $NF}'
 }
 
 # list of packages -- excluding some that can't be installed on Travis
 xs()
 {
   find \
     packages/xs                     \
+    packages/xs-extra               \
     -maxdepth 1 -mindepth 1 -type d \
-  | awk -F/ '{print $NF}'           \
-  | egrep -v '^(xenctrl|xapi-test-utils)'
+    | awk -F/ '{print $NF}'
 }
 
-# packages from xs-extra/ that we can compile
-XS_EXTRA='
-message-switch
-vncproxy
-wsproxy
-'
-
-# all of our packages that we can compile
-XS_ALL="$(xs) $XS_EXTRA"
-
-opam init -y local file://$PWD
-opam config exec -- opam install -y -j 4 $(upstream) $XS_ALL
-# Workaround to mark failed uninstall as error. We only test
-# the uninstall of the XS_ALL packages but not of the upstream packages.
-opam config exec -- opam remove -y $XS_ALL
-opam config exec -- opam install -y -j 4 $XS_ALL
-
+if [ ! -z "${EXTRA_REMOTES}" ]; then
+    opam remote add extra "$EXTRA_REMOTES"
```

Comment:
Where is the Opam repository initialised now? Maybe adding a comment would be helpful.